### PR TITLE
codeintel: Modify SoftDeleteOldDumps to delete old non-terminal uploads as well

### PR DESCRIPTION
**Pull request by efritz**
Fixes https://github.com/sourcegraph/sourcegraph/issues/17331.

### CODE REVIEW

File: enterprise/internal/codeintel/stores/dbstore/uploads.go [hunk 1]
```diff
@@ -503,3 +503,51 @@ const hardDeleteUploadByIDQuery = `
...
+	defer func() { err = tx.Done(err) }()
+
+	seconds := maxAge / time.Second
+	repositoryIDs, err := scanCounts(tx.Store.Query(ctx, sqlf.Sprintf(softDeleteOldUploadsQuery, now, seconds, now, seconds)))
+	if err != nil {
+		return 0, err
+	}
+
+	for repositoryID, numUpdated := range repositoryIDs {
+		if err := tx.MarkRepositoryAsDirty(ctx, repositoryID); err != nil {
```
Comment (Strum355) [inline enterprise/internal/codeintel/stores/dbstore/uploads.go:529]: just as a drive by comment, a brief explanation on this method for what happens after a repository is marked (who does what and when) would be super appreciated

File: enterprise/internal/codeintel/stores/dbstore/uploads.go [hunk 2]
```diff
@@ -503,3 +503,51 @@ const hardDeleteUploadByIDQuery = `
...
+			return 0, err
```
Comment (Strum355) [inline enterprise/internal/codeintel/stores/dbstore/uploads.go:530]: do we want to early bail on the first error or aggregated all errors? Feel the latter might be the better idea in the long run

**Review by Strum355 — APPROVED** [review]

File: enterprise/internal/codeintel/stores/dbstore/uploads.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/stores/dbstore/uploads.go:530]: It's all in a transaction so one error would likely imply there's an issue with all of them. If there's an error here the txn should rollback and it doesn't matter how many errors we've accumulated.

File: enterprise/internal/codeintel/stores/dbstore/uploads.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/stores/dbstore/uploads.go:530]: Oops, sorry I forgot to address this. Will do so in a new PR.

**Pull request merged**